code.rb

```diff
@@ -0,0 +1,178 @@
+module Enumerable
+  def my_each
+    size = self.size
+    index = 0
+
+    while index < size
+
+      yield(self[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    size = self.size
+    index = 0
+
+    while index < size
+
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+
+    my_each do |x|
+      new_array.push(x) if yield(x)
+    end
+
+    new_array
+  end
+
+  def my_all?(val = nil)
+    all_true = true
+
+    if block_given?
+
+      my_each do |x|
+        all_true = false unless yield(x)
+      end
+
+    elsif val
+
+      my_each do |x|
+        all_true = false unless x == val
+      end
+
+    else
+
+      my_each do |x|
+        all_true = false unless x
+      end
+
+    end
+
+    all_true
+  end
+
+  def my_any?(val = nil)
+    any = false
+
+    if block_given?
+      my_each do |x|
+        any = true if yield(x)
+      end
+
+    elsif val
+      my_each do |x|
+        any = true if x == val
+      end
+
+    else
+
+      my_each do |x|
+        any = true if x
+      end
+
+    end
+
+    any
+  end
+
+  def my_none?(val = nil)
```

Comment:
Similar issues as with the previous two methods.
Consider Classes and Regular Expressions